plugin/percona-pam-for-mysql/src/groups.c

```diff
@@ -41,23 +43,58 @@ struct groups_iter *groups_iter_new(const char *user_name)
   int error;
   struct groups_iter *it;
 
-  it= calloc(1, sizeof(struct groups_iter));
-  if (it == NULL)
-    return NULL;
+  if (gr_buf_size <= 0)
+  {
+    long gr_size_max, pw_size_max;
+    gr_size_max= sysconf(_SC_GETGR_R_SIZE_MAX);
```

Comment:
sysconf(_SC_GETGR_R_SIZE_MAX) is allowed to return -1 as a valid return value indicating no max size, this case needs to be handled